src/dmd/mars.d

```diff
@@ -245,21 +246,37 @@ private int tryMain(size_t argc, const(char)** argv)
         return EXIT_SUCCESS;
     }
 
-    if (global.params.mcpuUsage)
+    /*
+    Prints a supplied usage text to the console and
+    returns the exit code for the help usage page.
+
+    Returns:
+        `EXIT_SUCCESS` if no errors occurred, `EXIT_FAILURE` otherwise
+    */
+    static byte printHelpUsage(string help)
     {
-        import dmd.cli : CLIUsage;
-        auto help = CLIUsage.mcpu;
         printf("%.*s", help.length, &help[0]);
-        return EXIT_SUCCESS;
+        return global.errors ? EXIT_FAILURE : EXIT_SUCCESS;
     }
 
-    if (global.params.transitionUsage)
+    /*
+    Check for each usage flag whether it has been toggled
+    If so, print its help page and exit
+    */
+    static generateUsageChecks()
     {
-        import dmd.cli : CLIUsage;
-        auto help = CLIUsage.transitionUsage;
-        printf("%.*s", help.length, &help[0]);
-        return EXIT_SUCCESS;
+        string s;
+        foreach (n; ["mcpu", "transition", "check", "checkAction", "externStd"])
+        {
+            s ~= q{
+                if (global.params.}~n~q{Usage)
```

Comment:
Avoid using `global.params`, use local `params` instead. Initialize `const params` with `global.params`.

Reply:
Using `global.params` is ubiquotus in `tryMain`, e.g.

https://github.com/dlang/dmd/blob/bd43dcb01a487343320bf2c4771ce4dc8817fcba/src/dmd/mars.d#L162-L163

https://github.com/dlang/dmd/blob/bd43dcb01a487343320bf2c4771ce4dc8817fcba/src/dmd/mars.d#L236-L246

https://github.com/dlang/dmd/blob/bd43dcb01a487343320bf2c4771ce4dc8817fcba/src/dmd/mars.d#L296

(there's a lot more)

If I would refactor it in this PR here, you very likely would complain about the unrelated changes and mixing things in a PR.

Reply:
Hmm I still think that this could have been in a different PR, but it's super easy to do: 3cb064a5d